src/instructlab/model/serve_backend.py

```diff
@@ -95,6 +134,15 @@ def serve_backend(
             ctx.obj.config.serve.vllm.vllm_args.extend(
                 ["--tensor-parallel-size", str(gpus)]
             )
+        else:
+            tensor_parallel_size = check_tensor_parallel_size(
+                ctx.obj.config.serve.vllm.vllm_args, available_gpus
+            )
+            if tensor_parallel_size > 0:
```

Comment:
This is a bit confusing. How the code can be read is: "if tensor-parallel-size is set to > 0, then we always fail", which is not what we should impress. The actual logic is correct (if tensor-parallel-size is set to > available_gpus) is implemented correctly, but in a way that confuses the reader. How about?

```
tensor_parallel_size = get_tensor_parallel_size(vllm_args)
if tensor_parallel_size > available_gpus:
    logger.error(...)
    return
```

So you will have your `check` function (suggest rename into `get_`) returning the value as passed by the user (or the default value), while you validate if it's correct here inline.
```

Reply:
seems more clear..let me try it.